h/services/delete_group.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h.events import AnnotationEvent
+from h.models import Annotation
+from h import storage
+
+
+class DeleteGroupService(object):
+    def __init__(self, request):
+        self.request = request
+
+    def delete(self, group):
+        """
+        Deletes a group, its membership relations and all annotations in the
+        group.
+        """
+
+        self._delete_annotations(group)
```

Comment:
I think we should probably disallow deleting the `__world__` group here since we treat that specially in a few other places in the code.